### ISSUE TITLE
Add Sourcify support for proxies and beacons

### ISSUE THREAD

**Issue by 0xmichalis** [OpenZeppelin/openzeppelin-upgrades#819]
It'd be great for the plugin to be able to verify in Sourcify the proxies/beacons it deploys.

**Comment by frangio** [comment]
We use `hardhat-verify`. Does it support Sourcify? If not, that's probably a better place to start.

**Comment by haowang1013** [comment]
> We use `hardhat-verify`. Does it support Sourcify? If not, that's probably a better place to start.

Yes, `hardhat-verify` supports Sourcify, I was able to verify the implementation contract with it.
